HW05 Jerry Lee.Rmd

```diff
@@ -0,0 +1,116 @@
+
+dad_mom_tidy
+```
+
+## 2. Joining together CTRP data
+
+CTRP is a dataset I use in my research. CTRP screens cancer cells with various compounds and determines the Area Under the Dose Response Curve (AUC, a metric for how sensitive a cell line is to that compound). However, when you download the data, it comes in parts that requires a lot of id matching to be able to use reasonably easily. For example, the file that contains the AUC data doesn't tell you the cell line name or the type of cancer it is, it just gives an experiment ID number. If you wanted to know which cell line was which, you'd have to reference the experiment_info.csv file which gives you a cell line ID for each experiment, and then reference the the cancer_cell_line_file.csv to figure out what the name of that cell line actually is. 
+
+That is all to say, it would be much easier if those files were all together instead. You're goal is to join together the 5 csv together.
+
+```{r reading data #2}
+auc1<-read_csv("CTRP_files/AUC_1.csv")
+auc2<-read_csv("CTRP_files/AUC_2.csv")
+ccl<-read_csv("CTRP_files/cancer_cell_line_info.csv")
+cpd<-read_csv("CTRP_files/compound_info.csv")
+expt<-read_csv("CTRP_files/Experiment_info.csv")
+```
+```{r joining data}
+joined<-expt%>%
+  full_join(ccl,by="master_ccl_id")%>%
+  full_join(auc1,by=c("expt_id"="experiment_id"))%>%
+  full_join(auc2,by=c("expt_id"="experiment_id"))%>%
+  select(!c(area_under_curve.y,master_cpd_id.y))%>%
+  rename(area_under_curve=area_under_curve.x,master_cpd_id=master_cpd_id.x)%>%
+  full_join(cpd,by="master_cpd_id")
+```
+
+Then once those files are all together, you should have no problem answering the following questions (with graphs):
+
+* Which cancer type has the lowest AUC values to the compound "vorinostat"?
+
+```{r question 1}
+q1<-joined%>%
+  filter(cpd_name=="vorinostat")
+
+ggplot(q1,aes(y=fct_reorder(cancer_type,area_under_curve),x=area_under_curve))+
+  geom_boxplot()+
+  labs(title="Boxplot showing AUC for different cancer types to vorinostat",y="cancer type",x="AUC")
+```
+
+```{r answer 1}
+#autonomic_ganglia
+```
+
+* Which compound is the prostate cancer cell line 22RV1 most sensitive to? (For 22RV1, which compound has the lowest AUC value?)
+
+```{r question 2}
+q2<-joined%>%
+  filter(ccl_name=="22RV1")%>%
+  arrange(area_under_curve)%>%
+  slice(1:10) #too many compounds to graph, so I chose the ten that had the lowest AUC
+
+ggplot(q2,aes(y=fct_reorder(cpd_name,area_under_curve),x=area_under_curve))+
+  geom_col()+
+  labs(title="Barplot showing compounds with the ten lowest AUC to prostate cancer cell line 22RV1",y="compound",x="AUC")
+```
+
+```{r answer 2}
+#Leptomycin B
+```
+
+* For the 10 compounds that target EGFR, which of them has (on average) the lowest AUC values in the breast cancer cell lines?
+
+```{r question 3}
+q3<-joined%>%
+  filter(gene_symbol_of_protein_target=="EGFR"&cancer_type=="breast")
+
```

Comment:
Here's maybe one of the advantages of using the filter function here, like you did, rather than subset (which was what I used)--the ability to use the "&" argument to pass multiple filter criteria seems like a very straightforward approach. One disadvantage might be that then you have several filtered datasets rather than one that you just subset for each of your analyses. However, I'm not sure--this is just a guess.